benchmarks/README.md

```diff
@@ -0,0 +1,88 @@
+
+## Establishing Configurations
+
+To establish a set of configurations for a simulation, you should use ```config_generate.jl```, located in the ```scripts``` directory. This will read either the included ```main_config.toml``` or a user provided file, when the file name is provided as a command line argument, and distribute the configuration information to multiple tasks. 
+
+To use a custom TOML, please follow all the steps below:
+
+0. The user provided TOML must located in the ```src``` directory.
+1. For a given simulation (heat, brusselator, etc.) and intended architeture (cpu, cuda), include a group in the TOML as `[sim_name.arch]`, where `sim_name` and `arch` are respective names from the previous groups.
+2. Under a group, list all the parameters desired. This should be structured as either `param_name = [val1, val2, val3...]` or `param_name = val`.
+3. Highly recommended parameters to include are ```code_target```, which takes either ```CPUTarget``` or ```CUDATarget```.
+
+```config_generate.jl``` will generate the required directories automatically. If you were to include ```[heat.cpu]``` in your TOML, the resulting configuration would be in ```src/heat/heat_cpu.toml```.
+
+Please view ```main_config.toml``` as a guiding example on how to craft your own TOML. 
+
+**Warning**: ```config_generate.jl``` is not called automatically so it is up to you to run the script before launching benchmarks. 
+
+**Warning**: Do not call ```config_generate.jl``` for an actively running simulation since this may unexpectedly affect the running process.
+
+
+## Creating a Simulation File
+
+These benchmarks depend upon you to create the simulation files to be benchmarked. For a certain simulation named ```example```, include a file in ```src/example/example.jl```. Always start the file with the following, as it provides you access to helper functions located in ```src/helpers```.
+```julia
+using DrWatson
+@quickactivate :benchmarks
+```
+
+Additionally, this file should contain the following functions.
+
+0. ```setup_config```, which will take in a ```Dict{String, Any}``` with the provided parameter names pointing to that task's provided configuration values. It is up to you to take these values, process them and then organization them to be passed along.
+1. ```setup_benchmark```, which will create the Decapode and run ```eval(gensim())``` on it. Return the evaluated function.
+2. ```create_mesh```, which will create the mesh upon which the simulation will run and also initialize the initial conditions and any constants/parameters. Return the mesh, initial conditions and constants/parameters in that order.
+3. ```create_simulate```, which will take the generated mesh and evaluated function and run the simulate function. Return the resulting function.
+4. ```run_simulation```, which will take the resulting simulation function, initial conditions and constants/parameters and run the solve. Return the result of the solve.
```

Comment:
Instead of relying on functions with particular names, we could make a struct in Decapodes that is `SimulationConfig` that has these 4 functions as fields and then constructors can provide default implementations of these fields if it makes sense to go with a generic implementation.

That would give users a way of encapsulating a simulation into a reusable chunk for their benchmarks, which would live outside this repo.

Reply:
Yeah, that sounds good. This struct can live in the `src` and be brought into scope with the `@quickactive :benchmarks`. I'll also add a function field that takes in the file config information and organizes it into something the simulation can use throughout.